src/flb_upstream.c

```diff
@@ -258,15 +264,29 @@ int flb_should_proxy_for_host(const char *host, const char *proxy, const char *n
     }
 
     /* check the URL list in the NO_PROXY  */
-    char *no_proxy_url = strtok(no_proxy, ",");
-    while (no_proxy_url != NULL) {
-        if (strcmp(host, no_proxy_url) == 0) {
-            return FLB_FALSE;
+    ret = flb_slist_create(&no_proxy_list);
+    if (ret != 0) {
+        return FLB_TRUE;
+    }
+    ret = flb_slist_split_string(&no_proxy_list, no_proxy, ',', -1);
+    if (ret <= 0) {
+        return FLB_TRUE;
+    }
+    ret = FLB_TRUE;
+    mk_list_foreach(head, &no_proxy_list) {
+        e = mk_list_entry(head, struct flb_slist_entry, _head);
+         if (strcmp(host, e->str)) {
+            ret = FLB_FALSE;
+            break;
         }
-        no_proxy_url = strtok(NULL, ",");
     }
 
-    return FLB_TRUE;
+    /* clean up the resources. */
+    if (&no_proxy_list) {
```

Comment:
this check is invalid, ```no_proxy_list``` is a variable from the stack, if you want to check if it's empty can use mk_list_size(&no_proxy_list)  == 0 (to check if it has zero nodes linked)